site/content/docs/02-template-syntax.md

```diff
@@ -887,7 +887,140 @@ Local transitions only play when the block they belong to is created or destroye
 
 ### Animations
 
-TODO i can't remember how any of this works
+```sv
+animate:name
+```
+
+```sv
+animate:name={params}
+```
+
+```js
+animation = (node: HTMLElement, { from: DOMRect, to: DOMRect } , params: any) => {
+	delay?: number,
+	duration?: number,
+	easing?: (t: number) => number,
+	css?: (t: number, u: number) => string,
+	tick?: (t: number, u: number) => void
+}
+```
+
+```js
+DOMRect {
+	bottom: number,
+	height: number,
+	​​left: number,
+	right: number,
+	​top: number,
+	width: number,
+	x: number,
+	y:number
+}
+```
+
+---
+
+An animation is triggered when the contents of a [keyed each block](docs#Each_blocks) are re-ordered. Animations do not run when an element is removed, only when the each block's data is reordered. Animate directives must be on an element that is an *immediate* child of a keyed each block.
```

Comment:
`An animation is triggered when the elements of a [keyed each block](docs#Each_blocks) are re-ordered. Animations do not run for elements that are added or removed (like transitions are), only for those that are reordered. Animate directives must be on an element that is an *immediate* child of a keyed each block.`

Maybe something like this? I'm not thrilled about the parenthetical (like transitions are) shoehorned in there. I'm on the lookout for a better way to work that into the paragraph.

Reply:
Yeah, this works better. I think I made the assumption when writing this that people reading about `animate` would know about `transition` but each entry should stand alone.